### DISCUSSION TITLE
Two models running one by one

### DISCUSSION THREAD

**Discussion by bflo96** [OpenModelica/OpenModelica#12900] [General]
Hello everyone!

I'm constructing my problem like this in OpenModelica:

- Main package, called _Program_
- Two inner packages: _Hid_ and _Ther_
- In each package, there are several models to make different calculations. Since the calculation I need to perform is long, I divide it in "sections".
- Package Hid has a model named **CalculationHid**
- Package Ther has the models **InitializationTher** and **CalculationTher** 

The first model I want to run is independent from the other. When I want to run model **CalculationHid**, which has 571 equations/variables. I don't need **CalculationTher** to run it.

However, model **InitializationTher** and **CalculationTher** depend on some of the results I get in **CalculationHid**, namely current values of the resistors.

When I write (some lines are hidden):
```
model InitializationTher
   extends Program.Hid.CalculationHid;
   Ther.Flow A[10];
   Ther.Flow B[10];
   Ther.Flow C[10];
   equation
      for c in 1:10 loop
         A[c].Flow = abs(Ra[c].i);
         B[c].Flow = abs(Rb[c].i);
         C[c].Flow = abs(Rb[c].i);
      end for;
end InitializationTher;
```

This works just fine. The values saved in A[c].Flow etc are the results I got when I first ran  **CalculationHid**.

However, when I do, the values of the flowrates are different and result in a singular matrix in a later calculation of the code. Some values of the current come out as 1 or 0, which doesn't make any sense:

```
model CalculationTher 
      extends Ther.InitializationTher;
      Real var vA[10];
      Real var vB[10];
      Real var vC[10];
      equation
      for c in 1:10 loop
         vA = A[c].Flow * 20;
         vB = B[c].Flow * 10;
         vC = C[c].Flow * 10;
      end for;
end CalculationTher;
```

Could this be related to the initialization?
I figure that, even though **CalculationHid** does not need **CalculationTher** to run, the equations are solved in the same "context".

**I think what I'm missing is a way to actually run the models independently. How can I first run **CalculationHid** and then use the outputs of this model on **CalculationTher**, without doing it manually? Are there any guides on how to do this efficiently?**

Sorry in advance for any confusion on this thread and I can do my best to describe my problem if needed, as I probably did some oversimplifications to avoid too many details about my work.

Thank you!

**Comment by casella** [comment]
@bflo96 I am not sure I understand your problem exactly. As I understood
- You have a model called InitializationTher that runs successfully and computes some results that are needed by another model
- If you extend that model in CalculationTher, the same equations are not solved successfully

This is quite strange. If the equations in InitializationTher can be solved independently, the same equations inherited in CalculationTher should be solvable exactly in the same way, hence the generated code should be the same and have the same chances of success. There should be no need to run them independently. Unfortunately, to help you understanding what is going on here, we'd need to run the code ourselves and analyze it. Maybe you can produce a reduced MWE that you can share.

One way to use them independently is to put a when terminal() section in the first one, which writes the computed results to a file using the Modelica.Utilities functions, then define some parameters with fixed = false in the second model (without inheriting the equations) and add an initial equation section that reads the file and sets the values of those parameters for the simulation. A bit awkward, but it would run automatically. 

But, again, I would only see a reason to do this if those computations are very lengthy and you want to just read them multiple times without recomputing them every time. Other than that, if the first problem is independend from the second, when you inherit the code of the first problem, it should be solved exactly in the same way. Maybe you can check how it is solved in both cases using the [Equation-Based Debugger](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/debugger.html#the-equation-based-debugger) of OpenModelica.
